src/sidebar/services/annotations.js

```diff
@@ -59,7 +61,66 @@ export default function annotations(
     searchClient.get({ uri: uris, group: groupId });
   }
 
+  /**
+   * Apply changes for the given `annotation` from its draft in the store (if
+   * any) and return a new object with those changes integrated.
+   */
+  function applyDraftChanges(annotation) {
```

Comment:
This is essentially a reimplementation of the `Annotation` controllers local `updateModel` function:

https://github.com/hypothesis/client/blob/ec63961a743dc15e7112290816e19b38ac5d73e5/src/sidebar/components/annotation.js#L8-L19